setup.py

```diff
@@ -1,11 +1,27 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
-from setuptools import setup, Extension
+from setuptools import setup, Extension, find_packages
 from setup_cares import cares_build_ext
 import codecs
+import platform 
+import os
 
 __version__ = "1.0.0"
+kwargs = {}
+
+if platform.python_implementation() == 'CPython':
+    kwargs['ext_modules'] = [Extension('pycares._core',
+                                       sources = ['src/pycares.c'],
+                                       define_macros=[('MODULE_VERSION', __version__)]
+                                      )]
+
+# cffi module
+kwargs['setup_requires'] = ["cffi>=1.5.0"]
```

Comment:
Do you mean:
```python
if (platform.python_implementation() == 'PyPy' or
        os.environ.get('PYCARES_CFFI', '0') != '0'):
    # build cffi module, only when PyPy or environment cflags enable
else:
    # build original C ext, otherwise

```